File: 02_calculadora/calculadora.py
class Calculadora():
    def __init__(self, maxBat):
        self.bateria = 0
        self.maxBat = maxBat
    
    def sum(self, a,b):
        if(self.bateria > 0):
            self.bateria -= 1
            return a+b
        else: 
            return "Bateria insuficiente"
        
    def div(self, a, b):
        if(self.bateria > 0):
            self.bateria -= 1
            if (b == 0):
                return "Divisão por 0"
            return a/b
        else: 
            return "Bateria insuficiente"
    
    def mult(self, a, b):
        if self.bateria > 0:
            self.bateria -= 1
            return a*b
        else: 
            return "Bateria insuficiente"
    
    def charge(self, m):
        if (m<= self.maxBat):
            self.bateria += m
    
    def __str__(self):
      return "bateria: " + str(self.bateria) + "/" + str(self.maxBat)


calc = Calculadora(0)

print ("Digite init _maxCarga, charge _value, sum _a _b, div _a _b, mult _a _b, show ou end")
line = " "
while line != "end":
  line = input()
  ui = line.split(" ")
  
  if ui[0] == "end":
    break
  elif ui[0] =="show":
    print(calc)
  elif ui[0] == "init":
    calc.maxBat = int(ui[1])
  elif ui[0] == "charge":
    calc.charge(int(ui[1]))
  elif ui[0] == "sum":
    a = (int(ui[1]))
    b = (int(ui[2]))
    print(calc.sum(a,b))
  elif ui[0] == "div":
    a = (int(ui[1]))
    b = (int(ui[2]))
    print(calc.div(a,b))
  elif ui[0] == "mult":
    a = (int(ui[1]))
    b = (int(ui[2]))
    print(calc.mult(a,b))
  else:
    print ("Comando invalido")

File: 02_calculadora/calculadora.java
import java.util.Scanner;

class Calculadora {
	int bateria;
	int bateriaMax;
	
	public Calculadora(int bateria, int bateriaMax) {
		this.bateria = bateria;
		this.bateriaMax = bateriaMax;
	}
	
	public void soma(double a, double b) {
		if(this.bateria > 0) {
			this.bateria = this.bateria - 1;
			System.out.println(a + b);
		}else {
			System.out.println("fail: bateria insuficiente");
		}
	}
	
	public void divi(double a, double b) {
		if(this.bateria > 0) {
			this.bateria = this.bateria - 1;
			if(b == 0)
                System.out.println("fail: divisao por 0");
			System.out.println(a / b);
		}else {
			System.out.println("fail: bateria insuficiente");
		}
	}
	
	public void recarga(int m) {
		this.bateria += m;
		if(this.bateria > this.bateriaMax)
			this.bateria = bateriaMax;
	}
	
	public String toString() {
		return "bateria: " + this.bateria + "/" + this.bateriaMax;
	}
}

public class Controller{
	public static void main(String[] args) {
		
		Scanner scanner = new Scanner(System.in);
		Calculadora calculadora = new Calculadora(0, 0);
		
		while (true) {
			String line = scanner.nextLine();
			String[] ui = line.split(" ");
			
			if(ui[0].equals("end")){
				break;
			}else if(ui[0].equals("init")) {
				calculadora = new Calculadora(Integer.parseInt(ui[1]), Integer.parseInt(ui[2]));
			}else if(ui[0].equals("show")) {
				System.out.println(calculadora);
			}else if(ui[0].equals("charge")) {
				calculadora.recarga(Integer.parseInt(ui[1]));
			}else if(ui[0].equals("sum")) {
				calculadora.soma(Integer.parseInt(ui[1]), Integer.parseInt(ui[2]));
			}else if(ui[0].equals("div")) {
				calculadora.divi(Integer.parseInt(ui[1]), Integer.parseInt(ui[2]));
			}else {
				System.out.println("fail: comando invalido");
			}
		}
	}
}